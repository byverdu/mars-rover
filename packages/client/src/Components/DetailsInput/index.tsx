import React from 'react';
import './DetailsInput.scss';

export interface DetailsInputProps {
  title: string;
}

const DetailsInput: React.FC<DetailsInputProps> = ({title, children}) => (
  <details>
    <summary>{title}</summary>
    <div className="details-wrapper">
    <div className="details-styling">
      {children}
    </div>
    </div>
  </details>
);

export default DetailsInput
