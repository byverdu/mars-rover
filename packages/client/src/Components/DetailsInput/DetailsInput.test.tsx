import React from 'react';
import DetailsInput, { DetailsInputProps } from './index';
import enzyme, { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

const props: DetailsInputProps = {
  title: 'Set Plateau Size'
};

describe('<DetailsInput />', () => {
  const wrapper = mount(<DetailsInput {...props} children={<input/>} />);

  it('should create a snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should a title props', () => {
    expect((wrapper.props() as DetailsInputProps).title).toEqual('Set Plateau Size');
  });

  it('should have a <details> html tag', () => {
    expect(wrapper.find('details').length).toEqual(1);
  });
  
  it('should have a <summary> html tag with the title as text', () => {
    expect(wrapper.find('summary').length).toEqual(1);
    expect(wrapper.find('summary').text()).toEqual(props.title);
  });

  it('should render some children components', () => {
    expect(wrapper.props().children).toEqual(<input />);
  });
});
