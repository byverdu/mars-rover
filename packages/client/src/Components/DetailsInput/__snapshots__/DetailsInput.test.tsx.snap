// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsInput /> should create a snapshot 1`] = `
<DetailsInput
  title="Set Plateau Size"
>
  <details>
    <summary>
      Set Plateau Size
    </summary>
    <div
      className="details-wrapper"
    >
      <div
        className="details-styling"
      >
        <input />
      </div>
    </div>
  </details>
</DetailsInput>
`;
